### PR TITLE
Fix: useRbfForm compose callback

### PR DESCRIPTION
**Pull request by szymonlesisz**
## Description

Based on https://github.com/trezor/trezor-suite/pull/9193
Continuation of https://github.com/trezor/trezor-suite/pull/9046
Related to https://github.com/trezor/trezor-suite/issues/9045

NOTE: only **two last commits** are relevant, branch is based on https://github.com/trezor/trezor-suite/pull/9193 to avoid conflicts

1. `useRbfForm` works now only with complete data (selectedAccount is loaded + transaction have rbfParams). Both checks are done in `ChangeFee` component (props provider). Same as it is done in [sendform index](https://github.com/trezor/trezor-suite/blob/develop/packages/suite/src/views/wallet/send/index.tsx#L74-L82)
As result `state` of `useRbfForm` is memoized and never undefined (less conditions are needed)

2. `form/useCompose` composeRequest is unified with similar method of [useSendFormCompose hook.](https://github.com/trezor/trezor-suite/pull/9046) We are now one step closer to start using it in the `useSendForm` hook and get rid of  useSendFormCompose.
There is one additional [TODO regarding coinmarket hooks](https://github.com/trezor/trezor-suite/blob/35bd402662591f8f85b9af501b6d3545e37d795c/packages/suite/src/hooks/wallet/form/useCompose.ts#L24) and undefined `state` cc @LukasRada - do you plan to write some tests to your hooks, you need some help with that? it will be nice to know that we are not breaking things while refactoring.
As result less renders is done and useRbfForm tests are no longer require special "wait for render" case - test itself is now ~5 seconds shorter

### CODE REVIEW

File: packages/suite/src/hooks/wallet/useRbfForm.ts [hunk 1]
```diff
@@ -1,19 +1,25 @@
...
+} from '@suite-common/wallet-types';
 import { FormState, FeeInfo } from 'src/types/wallet/sendForm';
 import { useFees } from './form/useFees';
 import { useCompose } from './form/useCompose';
 import { selectCurrentTargetAnonymity } from 'src/reducers/wallet/coinjoinReducer';
 import { useBitcoinAmountUnit } from './useBitcoinAmountUnit';
 
 export type UseRbfProps = {
+    selectedAccount: SelectedAccountLoaded;
     tx: WalletAccountTransaction;
```
Comment (dahaca) [inline packages/suite/src/hooks/wallet/useRbfForm.ts]: no longer used

File: packages/suite/src/hooks/wallet/form/useCompose.ts [hunk 2]
```diff
@@ -137,8 +140,9 @@ export const useCompose = <TFieldValues extends FormState>({
                 clearErrors(composeErrors);
             }
 
-            // update feeLimit field if present (calculated from ethereum data size)
+            // update feeLimit field if present (calculated from ethereum data size)useEuseExchangeState(selectedAccount, fees, defaultValues)xchangeState(selectedAccount, fees, defaultValues)
```
Comment (dahaca) [inline packages/suite/src/hooks/wallet/form/useCompose.ts]: ?

File: packages/suite/src/hooks/wallet/useRbfForm.ts (hunk 1 above)
Comment (szymonlesisz) [reply packages/suite/src/hooks/wallet/useRbfForm.ts]: well, it is used :) before this change selectedAccount was taken from selector and could be undefined

or dou you mean `tx` prop? if so it is also used by ChangeFee component, to get get rbfParams and get/show current fee

File: packages/suite/src/hooks/wallet/form/useCompose.ts (hunk 2 above)
Comment (szymonlesisz) [reply packages/suite/src/hooks/wallet/form/useCompose.ts]: some additional unwanted paste

File: packages/suite/src/hooks/wallet/useRbfForm.ts (hunk 1 above)
Comment (szymonlesisz) [reply packages/suite/src/hooks/wallet/useRbfForm.ts]: i've moved `tx` prop to `ChangeFeeProps` and removed it from `UseRbfProps`
[ad0acac](https://github.com/trezor/trezor-suite/pull/9196/commits/ad0acaca4242035e227c54619c10ae3b18f15231)

File: packages/suite/src/components/suite/modals/TransactionDetail/components/ChangeFee/index.tsx [hunk 3]
```diff
@@ -101,14 +103,14 @@ const Red = styled.span`
...
     children?: React.ReactNode;
     showChained: () => void;
 }
 
-export const ChangeFee = (props: ChangeFeeProps) => {
+const ChangeFeeLoaded = (props: ChangeFeeProps) => {
     const contextValues = useRbf(props);
-    if (!contextValues.account) return null; // context without account, should never happen
-    const { tx } = props;
+    const { tx, chainedTxs, finalize } = props;
```
Comment (komret) [inline packages/suite/src/components/suite/modals/TransactionDetail/components/ChangeFee/index.tsx]:
```suggestion
    const { tx, chainedTxs, finalize, showChained, children } = props;
```

**Review by komret — APPROVED** [review]
Looks good. I haven't tested it, is there anything QA should try?

File: packages/suite/src/components/suite/modals/TransactionDetail/components/ChangeFee/index.tsx (hunk 3 above)
Comment (szymonlesisz) [reply packages/suite/src/components/suite/modals/TransactionDetail/components/ChangeFee/index.tsx]: [b431d9c](https://github.com/trezor/trezor-suite/pull/9196/commits/b431d9c8c60062e2374d546ad4e513a3b15f677d)

**Comment by szymonlesisz** [comment]
> is there anything QA should try?

i guess everything is covered by tests, they are working and they are working better than before :)

**Comment by szymonlesisz** [comment]
i've added another commit to increase unit tests coverage a little bit.

tests are not related to the changes introduced in this branch, new code added in `useRbfForm` hook is covered enough but for the overall coverage was not enough 🤷 (missing 0.09% for `branches` and 0.04% for `lines`)

**Pull request merged**